src/lakefs_spec/spec.py

```diff
@@ -474,6 +525,8 @@ def _api_path_type_to_info(path_type: str) -> Literal["file", "directory"]:
         path = cast(str, stringify_path(path))
         repository, ref, prefix = parse(path)
 
+        recursive = kwargs.pop("recursive", False)
+
         # Try lookup in dircache unless explicitly disabled by `refresh=True` kwarg
         use_dircache = True
```

Comment:
```suggestion
        use_dircache = not kwargs.pop("refresh", False)
```

then L532-534 can also go.

Reply:
Addressed in 926fb3be2d1eee23b75dfc9514e8570acff35519